roles/worker/tasks/main.yml

```diff
@@ -15,40 +15,74 @@
 #----------------------------------------------------------------------------
 ---
   - block:
-    - name: "Copy deployment.yaml"
+    # Set installation directory information
+    - name: Installation Information
+      set_fact:
+        product_dir: "{{ target }}/worker"
+        carbon_home: "{{ target }}/worker/{{ product_name }}-{{ product_version }}"
+
+    # WSO2 Distribution
+    - name: Create product directory
+      file:
+        path: "{{ product_dir }}"
+        state: directory
+        mode: 0755
+
+    - name: Copy wso2sp-worker service file
       template:
-        src: "carbon-home/conf/worker/deployment.yaml.j2"
-        dest: "{{ setup_path }}/{{ product_name }}/{{ product_version }}/conf/worker/deployment.yaml.j2"
+        src: wso2sp-worker.service.j2
+        dest: /etc/systemd/system/wso2sp-worker.service
+
+    - name: Stop wso2sp-worker as a service
+      systemd:
+        name: wso2sp-worker
+        state: stopped
+
+    - name: Remove existing setup
+      file:
+        path: "{{ carbon_home }}"
+        state: absent
 
-    - name: "Copy worker.sh"
+    - name: Unzip WSO2 Stream Processor Package
+      unarchive:
+        src: "{{ product_package_location }}/packs/{{ product_name }}-{{ product_version }}.zip"
+        dest: "{{ product_dir }}"
+        mode: u=rw,g=r,o=r
+
+#    - name: Copy JDBC Driver
```

Comment:
Shall we add the reason for this commented block as a descriptive comment?

Reply:
Done https://github.com/wso2/ansible-sp/pull/13/commits/7d7ba34d60829019a905f41b1f82e988b7802d52